shared/bundle_analysis/report.py

```diff
@@ -34,54 +33,58 @@ def size(self):
         return self.asset.size
 
     def modules(self):
-        return (
-            self.db_session.query(models.Module)
-            .join(models.Module.chunks)
-            .join(models.Chunk.assets)
-            .filter(models.Asset.id == self.asset.id)
-            .all()
-        )
+        with models.get_db_session(self.db_path) as session:
+            return (
+                session.query(models.Module)
+                .join(models.Module.chunks)
+                .join(models.Chunk.assets)
+                .filter(models.Asset.id == self.asset.id)
+                .all()
+            )
 
 
 class BundleReport:
     """
     Report wrapper around a single bundle (many of which can exist in a single analysis report).
     """
 
-    def __init__(self, bundle: models.Bundle):
+    def __init__(self, db_path: str, bundle: models.Bundle):
+        self.db_path = db_path
         self.bundle = bundle
-        self.db_session = SQLAlchemySession.object_session(self.bundle)
 
     @property
     def name(self):
         return self.bundle.name
 
     def asset_reports(self) -> Iterator[AssetReport]:
-        assets = (
-            self.db_session.query(models.Asset)
-            .join(models.Asset.session)
-            .join(models.Session.bundle)
-            .filter(models.Bundle.id == self.bundle.id)
-            .all()
-        )
-        return (AssetReport(asset) for asset in assets)
+        with models.get_db_session(self.db_path) as session:
```

Comment:
> > I'm not too familiar with this part of the codebase, but could we have modified the dunder del method for this class to close the session?
> 
> @ajay-sentry do you mean have the `AssetReport` and `BundleReport` create a DB session in `__init__` and close the session in `__del__`? Yeah that should work, though the indeterministic way of knowing when the garbage collection happens have me thinking that it's not ideal. I'm worried if there's a situation where there's no connections left before previous ones gets released.

@JerrySentry yep exactly what I meant. That's a good point, probably better to have them "ad-hoc" like you have here. Though, what happens now if the connections are full when we call this line?

Reply:
I think it would throw an error, not exactly sure. At least what happened before when never doing a close session was that after a few sessions were opened it would spit out an error and give up.

Reply:
If there's a way for us to test that scenario and make sure nothing catastrophic happens I think this lgtm :)

Reply:
Best way I could think of is to add a sleep in the `with`, then I made a bunch of requests at the same time. I didn't actually see any errors raised, what happened was the requests queued up till one of them was done. This is also not a true DB like the postgres we have, but it is just a mini SQLite file that is used for storing bundle stats of each commit; there shouldn't be anything catastrophic happening.